superset/db_engine_specs.py

```diff
@@ -104,15 +104,32 @@ def apply_limit_to_sql(cls, sql, limit, database):
             )
             return database.compile_sqla_query(qry)
         elif LimitMethod.FORCE_LIMIT:
-            no_limit = re.sub(r"""
+            sql_without_limit = cls.get_query_without_limit(sql)
+            return '{sql_without_limit} LIMIT {limit}'.format(**locals())
```

Comment:
this logic can generate sql like
```
SELECT id,
       name
FROM main.ab_permission
LIMIT 100
OFFSET 0 LIMIT 5000
```
which is wrong

Reply:
Which engine is this on?